chore: prepare and build aether containers

obdulia-losantos opened the pull request:
- Included `build_aether_containers.sh` for first run or occasional use.
- Named volume for database (windows requirement).
- Prettified `.env` file. :grimacing: 
- Updated `README`:
  - Aether description.
  - Removed UMS mention.
  - Rewritten authentication paragraph.
- Include aether icon in common module as static content.
- Collect static files in container setup (used by nginx container)
- **Execute the `release` step only once after tests passed and in the indicated cases.**

obdulia-losantos committed:
chore: build containers script

obdulia-losantos committed:
chore: simplify docker setup

obdulia-losantos committed:
chore(release): remove VERSION file

obdulia-losantos committed:
fix: move deploy condition to release script

admbtlr commented:
This version stuff probably won't work they way we want it - the Travis env var might include `release-` at the beginning of the version number, and also we want our release candidates to be for the _next_ version, not the last version.

I think maybe we just need to be a bit smarter about how we use the VERSION file instead of getting rid of it completely.

obdulia-losantos commented on scripts/release.sh:
```diff
@@ -24,27 +24,53 @@ release_app () {
+elif [[ $TRAVIS_BRANCH =~ ^release\-[0-9]+\.[0-9]+[\.0-9]*$ ]]
+then
+    # remove "release-" prefix and add "-rc" suffix
+    VERSION=${TRAVIS_BRANCH#"release-"}-rc
```
This line will remove the `release-` prefix on the branch name... would it make the trick for you @adamvert ?

Actually the only difference with the current way is that we lose the last number of the version. Now it's `0.9.4-rc` (if file content is `0.9.4`) with this change it'll be `0.9-rc` (if branch name is `release-0.9`).

admbtlr replied:
Right, that's the problem, and that's why we have the VERSION file after all - it would be better to have `0.9.x-rc`

obdulia-losantos replied:
Reintroduced file! :wink:

admbtlr replied:
Ideally, the script would write an incremented version to the VERSION file once it's done a release, so that subsequent RCs are correct.

obdulia-losantos replied:
If you don't mind I would prepare the AUTOMATIC incremented version in another PR...

obdulia-losantos committed:
fix: reintroduce VERSION file

admbtlr commented:
Yes, absolutely!

On Thu, Aug 16, 2018 at 6:16 PM, Obdulia Losantos <notifications@github.com>
wrote:

> *@obdulia-losantos* commented on this pull request.
> ------------------------------
>
> In scripts/release.sh
> <https://github.com/eHealthAfrica/aether/pull/329#discussion_r210657389>:
>
> >  }
>
> +# release version depending on TRAVIS_BRANCH/ TRAVIS_TAG
> +if [[ $TRAVIS_TAG =~ ^[0-9]+\.[0-9]+[\.0-9]*$ ]]
> +then
> +    VERSION=$TRAVIS_TAG
> +
> +elif [[ $TRAVIS_BRANCH =~ ^release\-[0-9]+\.[0-9]+[\.0-9]*$ ]]
> +then
> +    # remove "release-" prefix and add "-rc" suffix
> +    VERSION=${TRAVIS_BRANCH#"release-"}-rc
>
> If you don't mind I would prepare the AUTOMATIC incremented version in
> another PR...
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/eHealthAfrica/aether/pull/329#discussion_r210657389>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABuypOc4nwTeQEwyxpBVqSnu1IKiTlsDks5uRZrdgaJpZM4V_gvv>
> .
>



-- 

Adam Butler

Technical Team Lead

<http://www.ehealthafrica.org/>

eHealth & Information Systems Africa gGmbH
Oranienburger Str. 69
10117 Berlin
Germany

--

<https://web.facebook.com/EHealthAfrica>
<https://twitter.com/eHealth_africa>
<https://www.linkedin.com/company/1449060/>
<https://www.instagram.com/ehealthafrica/>
<https://www.youtube.com/channel/UCcgM_HhTcg5M9Ji3Of6QmUg>

obdulia-losantos committed:
fix(travis): skip deploy in tests matrix

obdulia-losantos committed:
fix: include favicon in common module

obdulia-losantos committed:
fix: collect static files in setup

obdulia-losantos committed:
fix(common): better in static/aether folder

shawnsarwar committed:
Merge branch 'develop' into feat-first-run

admbtlr committed:
fix: make the links in /forms work in local env (#333)

Without the port number in the `Host` header, the /formList endpoint in aether-odk will not include the port number. This means that ODK Collect will receive invalid form URLs and will not be able to download any forms.

obdulia-losantos committed:
Merge branch 'develop' into feat-first-run

obdulia-losantos committed:
docs: fix link in README

obdulia-losantos commented:
@adamvert should I change this PR to point to `release-0.9` branch too?

admbtlr commented:
Yes please!

obdulia-losantos committed:
Merge branch 'release-0.9' into feat-first-run

obdulia-losantos committed:
fix: remove UMS mentions

obdulia-losantos committed:
fix: prometheus settings

obdulia-losantos committed:
fix: tweaking media storage setup

admbtlr commented on docker-compose-base.yml:
```diff
@@ -264,21 +267,22 @@ services:
+
+      DEBUG: "true"
+      DJANGO_SECRET_KEY: ${UI_DJANGO_SECRET_KEY}
+      DJANGO_STORAGE_BACKEND: filesystem
```
I guess this can be taken out now that you removed `django-storages` from UI

obdulia-losantos replied:
Nop, it's mandatory to indicate it even if you are not using it:

https://github.com/eHealthAfrica/aether/blob/release-0.9/aether-common-module/aether/common/conf/settings.py#L249

admbtlr replied:
Ah, OK

obdulia-losantos committed:
docs: storage mandatory settings

obdulia-losantos commented:
@adamvert another concern? This time I brought my :hammer:

obdulia-losantos committed:
fix: docker-compose settings

obdulia-losantos committed:
fix: typo
